### ISSUE TITLE
failed to Load TinyYolo Onnx model in OnnxRuntime

### ISSUE THREAD

**Issue by shivSD** [microsoft/onnxruntime#5379]
**failed to Load TinyYolo Onnx model in OnnxRuntime**

  File "demo.py", line 6, in <module>
    sess = rt.InferenceSession("tiny_yolov2/model.onnx")
  File "/home/.local/lib/python3.6/site-packages/onnxruntime/capi/session.py", line 158, in __init__
    self._load_model(providers or [])
  File "/home/.local/lib/python3.6/site-packages/onnxruntime/capi/session.py", line 177, in _load_model
    self._sess.load_model(providers)
onnxruntime.capi.onnxruntime_pybind11_state.InvalidGraph: [ONNXRuntimeError] : 10 : INVALID_GRAPH : This is an invalid model. Error in Node:batchnorm : Required attribute 'consumed_inputs' is missing.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Ubuntu 18.04
- ONNX Runtime installed from (source or binary): Source 
- ONNX Runtime version: 1.5.1
- Python version: 3.6
- Visual Studio version (if applicable): NA
- GCC/Compiler version (if compiling from source):  gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04) 
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**To Reproduce**
- Describe steps/code to reproduce the behavior.
```
import numpy as np
import onnxruntime as rt
from PIL import Image,ImageDraw


sess = rt.InferenceSession("tiny_yolov2/model.onnx")
input_name = sess.get_inputs()[0].name
```

- Attach the ONNX model to the issue (where applicable) to expedite investigation.
https://www.cntk.ai/OnnxModels/tiny_yolov2/opset_1/tiny_yolov2.tar.gz

**Expected behavior**
should load the tinyyolo onnx model without any error

**Comment by shivSD** [comment]
Update:  i tried using [this](https://github.com/onnx/models/tree/master/vision/object_detection_segmentation/tiny-yolov2) onnx model and it loads without any error but does not give any output (no bounding box). Here is the my inference code. Please let me know if there is any wrong in the implementation

`import numpy as np
import onnxruntime as rt
from PIL import Image,ImageDraw


sess = rt.InferenceSession("tiny_yolov2/tinyyolov_remove.onnx")
input_name = sess.get_inputs()[0].name

img = Image.open('test.jpg')
img = img.resize((416, 416)) #for tiny_yolov2
X = np.asarray(img)
X = X.transpose(2,0,1)
X = X.reshape(1,3,416,416)
out = sess.run(None, {input_name: X.astype(np.float32)})
out = out[0][0]
numClasses = 20
anchors = [1.08, 1.19, 3.42, 4.41, 6.63, 11.38, 9.42, 5.11, 16.62, 10.52]

def sigmoid(x, derivative=False):
    return x*(1-x) if derivative else 1/(1+np.exp(-x))

def softmax(x):
    scoreMatExp = np.exp(np.asarray(x))
    return scoreMatExp / scoreMatExp.sum(0)

clut = [(0,0,0),(255,0,0),(255,0,255),(0,0,255),(0,255,0),(0,255,128),
  (128,255,0),(128,128,0),(0,128,255),(128,0,128),
  (255,0,128),(128,0,255),(255,128,128),(128,255,128),(255,255,0),
  (255,128,128),(128,128,255),(255,128,128),(128,255,128)]

label = ["aeroplane","bicycle","bird","boat","bottle",
         "bus","car","cat","chair","cow","diningtable",
         "dog","horse","motorbike","person","pottedplant",
          "sheep","sofa","train","tvmonitor"]

draw = ImageDraw.Draw(img)

for cy in range(0,13):
    for cx in range(0,13):
        for b in range(0,5):
            channel = b*(numClasses+5)
            tx = out[channel  ][cy][cx]
            ty = out[channel+1][cy][cx]
            tw = out[channel+2][cy][cx]
            th = out[channel+3][cy][cx]
            tc = out[channel+4][cy][cx]
            x = (float(cx) + sigmoid(tx))*32
            y = (float(cy) + sigmoid(ty))*32

            w = np.exp(tw) * 32 * anchors[2*b  ]
            h = np.exp(th) * 32 * anchors[2*b+1]

            confidence = sigmoid(tc)
            print(confidence)

classes = np.zeros(numClasses)
for c in range(0,numClasses):
    classes[c] = out[channel + 5 +c][cy][cx]
    classes = softmax(classes)
    detectedClass = classes.argmax()

if 0.5< classes[detectedClass]*confidence:
    color =clut[detectedClass]
    x = x - w/2
    y = y - h/2
    draw.line((x  ,y  ,x+w,y ),fill=color)
    draw.line((x  ,y  ,x  ,y+h),fill=color)
    draw.line((x+w,y  ,x+w,y+h),fill=color)
    draw.line((x  ,y+h,x+w,y+h),fill=color)

img.show()
img.save("result.png")`

**Comment by jcwchen** [comment]
Hi @shivSD,
I would say onnxruntime is good with this model (tiny yolov2) from onnx/models because it can be correctly inferred by onnxruntime. And there are some bugs in your inference code... Since I am not very familiar with tiny yolov2, I will suggest you to get the best help from tinyyolo team. IMO, it should not be an onnxruntime issue. Thank you.

Here are some possible bugs I found in the code
- The length of clut should be 20 instead of 19.
- If the model cannot detect the bounding box, you might need to lower the threshold.
- The last section of drawing is in a wrong place. It should be included into the previous nested loop.

A possible correction might be as follows:
```
import numpy as np
import onnxruntime as rt
from PIL import Image,ImageDraw


sess = rt.InferenceSession("tinyyolov2-8.onnx")
input_name = sess.get_inputs()[0].name

img = Image.open('test.jpg')
img = img.resize((416, 416)) #for tiny_yolov2
X = np.asarray(img)
X = X.transpose(2,0,1)
X = X.reshape(1,3,416,416)
out = sess.run(None, {input_name: X.astype(np.float32)})
out = out[0][0]
numClasses = 20
anchors = [1.08, 1.19, 3.42, 4.41, 6.63, 11.38, 9.42, 5.11, 16.62, 10.52]

def sigmoid(x, derivative=False):
    return x*(1-x) if derivative else 1/(1+np.exp(-x))

def softmax(x):
    scoreMatExp = np.exp(np.asarray(x))
    return scoreMatExp / scoreMatExp.sum(0)

clut = [(0,0,0),(255,0,0),(255,0,255),(0,0,255),(0,255,0),(0,255,128),
        (128,255,0),(128,128,0),(0,128,255),(128,0,128),
        (255,0,128),(128,0,255),(255,128,128),(128,255,128),(255,255,0),
        (255,128,128),(128,128,255),(255,128,128),(128,255,128),(128,255,128)]

label = ["aeroplane","bicycle","bird","boat","bottle",
         "bus","car","cat","chair","cow","diningtable",
         "dog","horse","motorbike","person","pottedplant",
          "sheep","sofa","train","tvmonitor"]

draw = ImageDraw.Draw(img)
threshold = 0.1

for cy in range(0,13):
    for cx in range(0,13):
        for b in range(0,5):
            channel = b*(numClasses+5)
            tx = out[channel  ][cy][cx]
            ty = out[channel+1][cy][cx]
            tw = out[channel+2][cy][cx]
            th = out[channel+3][cy][cx]
            tc = out[channel+4][cy][cx]
            x = (float(cx) + sigmoid(tx))*32
            y = (float(cy) + sigmoid(ty))*32

            w = np.exp(tw) * 32 * anchors[2*b  ]
            h = np.exp(th) * 32 * anchors[2*b+1]

            confidence = sigmoid(tc)
            print(confidence)

            classes = np.zeros(numClasses)
            for c in range(0,numClasses):
                classes[c] = out[channel + 5 +c][cy][cx]
                classes = softmax(classes)
                detectedClass = classes.argmax()

            if threshold < classes[detectedClass]*confidence:
                color =clut[detectedClass]
                x = x - w/2
                y = y - h/2
                draw.line((x  ,y  ,x+w,y ),fill=color)
                draw.line((x  ,y  ,x  ,y+h),fill=color)
                draw.line((x+w,y  ,x+w,y+h),fill=color)
                draw.line((x  ,y+h,x+w,y+h),fill=color)

img.show()
img.save("result.png")
```
